homeassistant/components/mvglive/sensor.py

```diff
@@ -140,88 +140,109 @@ def extra_state_attributes(self):
         return attr
 
     @property
-    def icon(self):
+    def icon(self) -> str | None:
         """Icon to use in the frontend, if any."""
         return self._icon
 
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit this state is expressed in."""
-        return UnitOfTime.MINUTES
-
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Get the latest data and update the state."""
-        self.data.update()
+        await self.data.update()
         if not self.data.departures:
             self._state = "-"
             self._icon = ICONS["-"]
         else:
-            self._state = self.data.departures[0].get("time", "-")
-            self._icon = ICONS[self.data.departures[0].get("product", "-")]
+            self._state = self.data.departures[0].get("time_in_mins", "-")
+            self._icon = self.data.departures[0].get("icon", ICONS["-"])
+
+
+def _get_minutes_until_departure(departure_time: int) -> int:
+    """Calculate the time difference in minutes between the current time and a given departure time.
+
+    Args:
+        departure_time: Unix timestamp of the departure time, in seconds.
+
+    Returns:
+        The time difference in minutes, as an integer.
+
+    """
+    current_time = datetime.now()
+    departure_datetime = datetime.fromtimestamp(departure_time)
+    time_difference = (departure_datetime - current_time).total_seconds()
```

Comment:
Naive datetimes are the root of all evil, can we use timezone aware datetimes instead, e.g.:
```py
current_time = dt_util.utcnow()
departure_datetime = dt_util.utc_from_timestamp(departure_time)
time_difference = (departure_datetime - current_time).total_seconds()
return int(time_difference / 60.0)
```